### PR TITLE
Feature/custom user attributes

### PR DESCRIPTION
**Pull request by bsekar**

### CODE REVIEW

File: server/src/main/resources/org/cloudfoundry/identity/uaa/db/postgresql/V4_101__AddCustomAttributesToUser.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE users ADD COLUMN custom_attributes TEXT DEFAULT NULL;
```
Comment (arjunnayak) [inline server/src/main/resources/org/cloudfoundry/identity/uaa/db/postgresql/V4_101__AddCustomAttributesToUser.sql:1]: could we use JSON type instead? it enables validation and better/more efficient parsing: https://www.postgresql.org/docs/12/datatype-json.html

File: server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java [hunk 2]
```diff
@@ -216,7 +212,13 @@ public ScimUser create(final ScimUser user, String zoneId) {
...
                 ps.setNull(17, Types.BIGINT);
                 ps.setNull(18, Types.BIGINT);
-                ps.setString(19, user.getPassword());
+                if(user.getCustomAttributes() == null) {
+                    ps.setNull(19, Types.LONGVARCHAR);
+                } else {
+                    ps.setString(19, JsonUtils.writeValueAsString(user.getCustomAttributes()));
+                }
+                ps.setString(20, user.getPassword());
+
```
Comment (arjunnayak) [inline server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java:219]: do we need to handle user update?

Comment (bsekar) [reply server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java:219]: Though this feature is to get custom attributes in POST /Users and the SAML sso , we should make sure the PUT doesn't overwrite what's already there. I believe the PreparedStatement will update only the properties provided and should leave the customAttributes unmodified during an Update.

Support for custom attributes in user update can be a follow up story?

Comment (arjunnayak) [reply server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java:219]: Put was handled in recent commit

**Comment by arjunnayak** [comment]
Ran unit and integration tests locally after downloading chrome webdriver via brew and adding some hosts to /etc/hosts. Only one IT failed: `test_LDAP_Custom_User_Attributes_In_ID_Token`. This error doesn't have to do with our custom attributes though, more likely ldap server setup. Error:
```
401 : [{"error":"unauthorized","error_description":"localhost:389; nested exception is javax.naming.CommunicationException: localhost:389 [Root exception is java.net.ConnectException: Connection refused (Connection refused)]"}]
org.springframework.web.client
```

As a result, I will merge this.

**Pull request merged**